lib/chef/resource/windows_certificate.rb

```diff
@@ -277,7 +277,9 @@ def acl_script(hash)
         def convert_pem(ext)
           out = case ext
                 when ".crt", ".cer", ".der"
-                  powershell_out("openssl x509 -text -inform DER -in #{new_resource.source} -outform PEM")
+                  command = "openssl x509 -text -in #{new_resource.source} -outform PEM"
```

Comment:
Is this something we can just do directly in ruby vs. shelling out?

Reply:
Yes, In ruby there is an `openssl` library which can easily perform all such operations.
```
OpenSSL::X509::Certificate.new(File.read(base64)).to_pem
=> CERTIFICATE CONTENT
```
But It doesn't seems to be working for all the cases. Like for DER(and even PFX, with keys) formats, it does not give the expected results:
```
OpenSSL::X509::Certificate.new(File.read(der))
=> OpenSSL::X509::CertificateError (not enough data)
```

Not sure, but may be due to this reason, openssl utility via powershell was preferred here.

But if it is permitted then _we can totally skip all shellouts, `openssl` and conversion operations from here_ and can use Crypt API in `win32-certstore`.

Also, due to this implementation we are importing all the certificates in PEM format into the certificate store.  and we've seen its drawback like data loss(private keys of PFX certificates).

Reply:
Hi @tas50 , I've just realized and found this, for DER(binary) certificates, we can get the contents by using **File.binread** :)